src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -11,14 +11,14 @@
 
 import java.util.List;
 
-public class PineconeFutureDataPlaneClient implements PineconeDataPlaneInterface<ListenableFuture<UpsertResponse>,
+public class AsyncIndex implements PineconeDataPlaneInterface<ListenableFuture<UpsertResponse>,
```

Comment:
Renamed PineconeFutureDataPlaneClient to AsyncIndex

Reply:
I like it. I think we should also update the `PineconeDataPlaneInterface` to `IndexInterface` to align with your updates.